### PR TITLE
Feature/avx 67 add debug target

### PR DESCRIPTION
**Pull request by rabad-matternet**
Added debug target to makefile.

To make a debug version:
`make debug`

No longer need to edit ntripclient.c to create a debug version.

Also added PHONY targets to makefile.

### CODE REVIEW

**Comment by zalice** [comment]
@rabad-matternet could you please populate the description of this PR?

File: makefile [hunk 1]
```diff
@@ -9,16 +9,28 @@ LIBS = -lwsock32
 else
 OPTS = -std=gnu99 -Wall -W -O3 
 endif
+DEBUG_OPTS = -DNTRIPCLIENT_DEBUG_LOG
```
Comment (zalice) [inline makefile:12]: Does this change the process of compilation of ntripclient with debug enabled?

Comment (rabad-matternet) [reply makefile:12]: Yes and no lol

The regular compilation is still built with:

`make`

Previous to this change, we built the debug version by #define NTRIPCLIENT_DEBUG_LOG in ntripclient.c.  With this change, we simply make the debug version with:

`make debug`

There is no longer the need to edit the .c file to create the debug version.

**Review by AydenChubbic — APPROVED** [review]

**Comment by rabad-matternet** [comment]
> @rabad-matternet could you please populate the description of this PR?

Done!

File: makefile (hunk 1 above)
Comment (zalice) [reply makefile:12]: ah, got it. could you please add this to the readme?

**Review by zalice — APPROVED** [review]

File: makefile (hunk 1 above)
Comment (rabad-matternet) [reply makefile:12]: Done!

**Pull request merged**